File: NOWCODER/剑指Offer/用两个栈实现队列/用两个栈实现队列.java
import java.util.Stack;

public class Solution {
    Stack<Integer> stack1 = new Stack<Integer>();
    Stack<Integer> stack2 = new Stack<Integer>();
    
    public void push(int node) {
        stack1.push(node);
    }
    
    public int pop() {
    	while(!stack1.isEmpty()){
    		stack2.push(stack1.pop());
    	}
    	int res = stack2.pop();
    	while(!stack2.isEmpty()){
    		stack1.push(stack2.pop());
    	}
    	return res;
    }
}

File: NOWCODER/剑指Offer/用两个栈实现队列/用两个栈实现队列.py
# -*- coding: utf-8 -*-
"""
Created on Wed Aug 23 22:40:51 2017

@author: Luohao
"""

"""
用两个栈来实现一个队列，完成队列的Push和Pop操作。
队列中的元素为int类型。
"""

#20 ms 5768K 
class Solution:
    def __init__(self):
        self.stack1 = []
        self.stack2 = []
    def push(self, node):
        # write code here
        self.stack1.append(node)
    def pop(self):
        if self.stack2 == []:
            while self.stack1 != []:
                self.stack2.append(self.stack1.pop())
        return self.stack2.pop()
        # return xx